nuc_morph_analysis/analyses/linear_regression/linear_regression_workflow.py

```diff
@@ -0,0 +1,261 @@
+        all_input = data[cols].reset_index(drop=True).values
+        all_target = data[target].values
+
+        if this_alpha == 0:
+            # linear regression if alpha == 0
+            clf = linear_model.LinearRegression()
+        else:
+            clf = linear_model.Lasso(alpha=this_alpha)
+
+        # normalize input features
+        model = make_pipeline(StandardScaler(), clf)
+
+        # run permutation test
+        score, permutation_scores, pvalue = permutation_test_score(
+            model, all_input, all_target, random_state=random_state, cv=5, n_permutations=500,
+        )
+
+        # break if permutation score is less than linear regression value (max possible)
+        # with a tolerance
+        # or if p_value > 0.05
+        rounded_permutation_score = round(score, 2)
+        if alpha_ind == 0:
+            max_val = rounded_permutation_score
+        if abs(rounded_permutation_score - max_val) > tol or (pvalue > 0.05):
+            break
+
+        # if relatively equal to linear regression value, then continue
+        # save permutation score and p_value to dictionary
+        all_perms['score'].append(score)
+        all_perms['perm_score_mean'].append(permutation_scores.mean())
+        all_perms['perm_score_std'].append(permutation_scores.std())
+        all_perms['p_value'].append(pvalue)
+        all_perms['alpha'].append(this_alpha)
+
+        # run cross validate to get model coefficients
+        cv_model = cross_validate(
+            model,
+            all_input,
+            all_target,
+            cv=RepeatedKFold(n_splits=5, n_repeats=20, random_state=random_state),
+            return_estimator=True,
+            n_jobs=2,
+            scoring=[
+                "r2",
+                "explained_variance",
+                "neg_mean_absolute_error",
+                "max_error",
+                "neg_mean_squared_error",
+                "neg_mean_absolute_percentage_error",
+            ],
+            return_train_score=True,
+        )
+
+        # Save test r^2 and test MSE to dataframe
+        range_test_scores = [round(i, 2) for i in cv_model["test_r2"]]
+        range_errors = [
+            round(i, 2) for i in cv_model["test_neg_mean_squared_error"]
+        ]
+        test_sc = pd.DataFrame()
+        test_sc[r"Test r$^2$"] = range_test_scores
+        test_sc["Test MSE"] = range_errors
+        test_sc["alpha"] = this_alpha
+        all_test_sc.append(test_sc)
+
+        # Save coeffs to dataframe
+        coefs = pd.DataFrame(
+            [model[1].coef_ for model in cv_model["estimator"]], columns=cols
+        )
+
+        all_coefs.append(coefs)
+        all_coefs = pd.concat(all_coefs, axis=0).reset_index(drop=True)
+        all_coefs["alpha"] = this_alpha
+        all_coef_alpha.append(all_coefs)
+
+    # Get test scores for all alpha
+    all_test_sc = pd.concat(all_test_sc, axis=0).reset_index(drop=True)
+    all_test_sc["Test MSE"] = -all_test_sc["Test MSE"]
```

Comment:
why make these values negative?

Reply:
the actual score in sklearn is `neg_mean_squared_error`. We want MSE, so I just take the negative of that